tests/src/core/geometry/testqgsnurbscurve.cpp

```diff
@@ -0,0 +1,654 @@
+    void toFromWkt();
+    void toFromWktZ();
+    void toFromWktM();
+    void toFromWktZM();
+    void toFromWktWithWeights();
+    void toFromWktWithKnots();
+    void toFromWktEmpty();
+    void toFromWkb();
+    void wkbCompatibilityWithSFCGAL();
+    void asGeometry();
+    void cast();
+};
+
+void TestQgsNurbsCurve::emptyConstructor()
+{
+  QgsNurbsCurve curve;
+
+  QVERIFY( curve.isEmpty() );
+  QCOMPARE( curve.numPoints(), 0 );
+  QCOMPARE( curve.vertexCount(), 0 );
+  QCOMPARE( curve.degree(), 0 );
+  QVERIFY( curve.controlPoints().isEmpty() );
+  QVERIFY( curve.knots().isEmpty() );
+  QVERIFY( curve.weights().isEmpty() );
+  QVERIFY( !curve.is3D() );
+  QVERIFY( !curve.isMeasure() );
+  QCOMPARE( curve.wkbType(), Qgis::WkbType::NurbsCurve );
+  QCOMPARE( curve.wktTypeStr(), QString( "NurbsCurve" ) );
+  QCOMPARE( curve.geometryType(), QString( "NurbsCurve" ) );
+  QCOMPARE( curve.dimension(), 1 );
+}
+
+void TestQgsNurbsCurve::constructorWithParams()
+{
+  QVector<QgsPoint> controlPoints { QgsPoint( 0, 0 ), QgsPoint( 1, 1 ), QgsPoint( 2, 0 ), QgsPoint( 3, 1 ) };
+  int degree = 2;
+  QVector<double> knots { 0, 0, 0, 1, 2, 2, 2 };
+  QVector<double> weights { 1, 1, 1, 1 };
+
+  QgsNurbsCurve curve( controlPoints, degree, knots, weights, false );
+
+  QVERIFY( !curve.isEmpty() );
+  QCOMPARE( curve.degree(), 2 );
+  QCOMPARE( curve.controlPoints().size(), 4 );
+  QCOMPARE( curve.knots().size(), 7 );
+  QCOMPARE( curve.weights().size(), 4 );
+  QVERIFY( !curve.isClosed() );
+}
+
+void TestQgsNurbsCurve::constructorDegree1()
+{
+  // Linear NURBS curve
+  QVector<QgsPoint> controlPoints { QgsPoint( 0, 0 ), QgsPoint( 10, 10 ) };
+  int degree = 1;
+  QVector<double> knots { 0, 0, 1, 1 };
+  QVector<double> weights { 1, 1 };
+
+  QgsNurbsCurve curve( controlPoints, degree, knots, weights );
+
+  QCOMPARE( curve.degree(), 1 );
+  QCOMPARE( curve.controlPoints().size(), 2 );
+  QCOMPARE( curve.knots().size(), 4 );
+}
+
+void TestQgsNurbsCurve::constructorDegree2()
+{
+  // Quadratic NURBS curve
+  QVector<QgsPoint> controlPoints { QgsPoint( 0, 0 ), QgsPoint( 5, 10 ), QgsPoint( 10, 0 ) };
+  int degree = 2;
+  QVector<double> knots { 0, 0, 0, 1, 1, 1 };
+  QVector<double> weights { 1, 1, 1 };
+
+  QgsNurbsCurve curve( controlPoints, degree, knots, weights );
+
+  QCOMPARE( curve.degree(), 2 );
+  QCOMPARE( curve.controlPoints().size(), 3 );
+  QCOMPARE( curve.knots().size(), 6 );
+}
+
+void TestQgsNurbsCurve::constructorDegree3()
+{
+  // Cubic NURBS curve (Bézier)
+  QVector<QgsPoint> controlPoints { QgsPoint( 0, 0 ), QgsPoint( 3, 10 ), QgsPoint( 7, 10 ), QgsPoint( 10, 0 ) };
+  int degree = 3;
+  QVector<double> knots { 0, 0, 0, 0, 1, 1, 1, 1 };
+  QVector<double> weights { 1, 1, 1, 1 };
+
+  QgsNurbsCurve curve( controlPoints, degree, knots, weights );
+
+  QCOMPARE( curve.degree(), 3 );
+  QCOMPARE( curve.controlPoints().size(), 4 );
+  QCOMPARE( curve.knots().size(), 8 );
+}
+
+void TestQgsNurbsCurve::constructorDegree4()
```

Comment:
not clear to me if constructorDegree4() and constructorDegree5() exercice code paths not already exercised by above tests